### PR TITLE
dev-python/configshell-fb: enable py3.11

### PR DESCRIPTION
**Pull request by arkamar**
- update EAPI 7 -> 8
- enable py3.11
- migrate to PEP517
- use github source instead of pypi
- apply two patches from upstream fixing deprecated function in py3.11
- add simple smoke test from examples, the proper test suite is missing
- indent metadata

### CODE REVIEW

File: dev-python/configshell-fb/configshell-fb-1.1.29-r1.ebuild [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+PYTHON_COMPAT=( python3_{9..11} )
+DISTUTILS_USE_PEP517=setuptools
+inherit distutils-r1
+
+DESCRIPTION="A Python library for building configuration shells"
+HOMEPAGE="https://github.com/open-iscsi/configshell-fb"
```
Comment (mgorny) [inline dev-python/configshell-fb/configshell-fb-1.1.29-r1.ebuild:11]:
```suggestion
HOMEPAGE="
	https://github.com/open-iscsi/configshell-fb/
	https://pypi.org/project/configshell-fb/
"
```

File: dev-python/configshell-fb/configshell-fb-1.1.29-r1.ebuild [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+RDEPEND="
+	dev-python/pyparsing[${PYTHON_USEDEP}]
+	dev-python/six[${PYTHON_USEDEP}]
+	dev-python/urwid[${PYTHON_USEDEP}]
+"
+
+PATCHES=(
+	"${FILESDIR}/${P}-replace-getargspec-1.patch"
+	"${FILESDIR}/${P}-replace-getargspec-2.patch"
```
Comment (mgorny) [inline dev-python/configshell-fb/configshell-fb-1.1.29-r1.ebuild:29]: Perhaps concatenate them to save space, or fetch via `SRC_URI`.

Comment (arkamar) [reply dev-python/configshell-fb/configshell-fb-1.1.29-r1.ebuild]: They are fetched via `SRC_URI` now.

**Comment by arkamar** [comment]
I have appended `Closes:` trailer with link to b.g.o to the commit mesage.

**Review by mgorny — APPROVED** [review]
LGTM. Thanks!

**Pull request closed without merging**